legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/schemaevolution/SchemaEvolution.java

```diff
@@ -174,9 +205,31 @@ private void evolveDataType(Field newField, Field mainDataField, Dataset mainDat
     {
         if (!mainDataField.equals(newField))
         {
-            operations.add(Alter.of(mainDataset, Alter.AlterOperation.CHANGE_DATATYPE, newField, Optional.empty()));
+            if (!mainDataField.type().equals(newField.type()))
+            {
+                operations.add(Alter.of(mainDataset, Alter.AlterOperation.CHANGE_DATATYPE, newField, Optional.empty()));
+                modifiedFields.add(newField);
+            }
+            // todo : tinyint(16) -> int (8) and nullability changes. does above statement handle both ?
+            else if (mainDataField.nullable() != newField.nullable())
+            {
+                alterColumnWithNullable(newField, mainDataset, operations, modifiedFields);
+            }
+        }
+    }
+
+    private void alterColumnWithNullable(Field newField, Dataset mainDataset, List<Operation> operations, List<Field> modifiedFields)
+    {
+        if (schemaEvolutionCapabilitySet.contains(SchemaEvolutionCapability.COLUMN_NULLABILITY_CHANGE))
+        {
+            operations.add(Alter.of(mainDataset, Alter.AlterOperation.NULLABLE_COLUMN, newField, Optional.empty()));
+            newField.withNullable(true);
```

Comment:
1) If main field is nullable and staging field is not nullable, Are we creating an unnecessary Alter here ?
2)  newField.withNullable(true) should have been done before